lambdas/user_access_patterns/lambda_for_createUser/index.js

```diff
@@ -1,6 +1,58 @@
-const AWS = require('aws-sdk');
-exports.handler = async (event) => {
-    // TODO implement
-    const response = (success, reason, data) => { return { success: success, reason: reason, data: data } }
-    return response(true, null, null);
+const AWS = require("aws-sdk");
+const crypto = require("crypto");
+//AWS.config.loadFromPath("../../../keys.json");
+//Use the below code for local setup
+AWS.config.update({
+  region: "local",
+  endpoint: "http://localhost:8000",
+});
+
+const dynamo = new AWS.DynamoDB.DocumentClient();
+function response(statusCode, message) {
+  return {
+    statusCode: statusCode,
+    body: JSON.stringify(message),
+  };
+}
+
+function hmac_hash(secretKey, password) {
+  const hashingSecret = secretKey;
+  const plainText = password;
+  const hashedStr = crypto
+    .createHmac("sha256", hashingSecret)
+    .update(plainText)
+    .digest("hex");
+
+  return hashedStr;
+}
+
+module.exports.createUser = (event, ctx, callback) => {
+  const reqBody = JSON.parse(event.body);
+  const user = {
+    u_name: reqBody.u_name,
+    u_age: reqBody.u_age,
+    u_thumbnail: reqBody.u_thumbnail,
+    u_is_verified: true,
+    u_provider: reqBody.u_provider,
+    u_password: hmac_hash(reqBody.u_email, reqBody.u_password),
+    u_space_used: 0,
+    u_user_type: "default",
+  };
+
+  return dynamo
+    .put({
+      TableName: "V-Transfer",
+      Key:{
+        PK:`USER#${reqBody.u_email}`,
+        SK:`#METADATA#${reqBody.u_email}`
+      }
+      Item: user,
+    })
+    .promise()
+    .then(() => {
+      callback(null, response(201, `User ${user.u_name} created successfully`));
```

Comment:
any specific reason for using callback??, why can't we simply return the result??

Reply:
Just convention for  keeping us safe from problems and errors.

Reply:
how can callback serve that purpose, as far I know lambdas are meant to return the response in the desired format and not depend on any other callbacks or context.

Reply:
I suggest not using callback and context, keep things simple and just return the response.

Reply:
It will be helpful to debug lambda if any error persist in future.

Reply:
have you tested the code on aws??